src/actions/profile.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable import/prefer-default-export */
+import axios from 'axios';
+import {
+  GET_PROFILE,
+  PROFILE_ERROR,
+  GET_FOLLOWERS,
+  GET_FOLLOWING,
+  GET_USER_ARTICLES,
+  UPDATE_PROFILE_SUCCESS,
+  UPDATE_PROFILE_FAIL
+} from './types';
+import setAuthToken from '../utils/setAuthToken';
+import { setContentType } from '../utils/setContentType';
+import { setAlert } from './alert';
+
+const config = setContentType('multipart/form-data');
+
+const performAction = (type, payload) => ({
+  type,
+  payload
+});
+
+export const getCurrentProfile = () => async (dispatch) => {
+  if (localStorage.token) {
+    setAuthToken(`Bearer ${localStorage.token}`);
+  }
+
+  try {
+    const res = await axios.get('/api/users/profile');
+    dispatch(performAction(GET_PROFILE, res.data));
+  } catch (error) {
+    dispatch(performAction(PROFILE_ERROR, error.response));
+  }
+};
+
+export const getUserFollowers = () => async (dispatch) => {
+  if (localStorage.token) {
+    setAuthToken(`Bearer ${localStorage.token}`);
+  }
+
+  try {
+    const res = await axios.get('/api/users/followers');
+
+    dispatch(performAction(GET_FOLLOWERS, res.data));
+  } catch (error) {
+    dispatch(performAction(PROFILE_ERROR, error.response));
+  }
+};
+
+export const getUserFollowing = () => async (dispatch) => {
+  if (localStorage.token) {
+    setAuthToken(`Bearer ${localStorage.token}`);
+  }
+
+  try {
+    const res = await axios.get('/api/users/following');
+
+    dispatch(performAction(GET_FOLLOWING, res.data));
+  } catch (error) {
+    dispatch(performAction(PROFILE_ERROR, error.response));
+  }
+};
+
+export const getUserArticles = () => async (dispatch) => {
+  if (localStorage.token) {
+    setAuthToken(`Bearer ${localStorage.token}`);
+  }
+
+  try {
+    const res = await axios.get('/api/users/articles');
+
+    dispatch(performAction(GET_USER_ARTICLES, res.data));
+  } catch (error) {
+    dispatch(performAction(PROFILE_ERROR, error.response));
+  }
+};
+
+export const updateProfile = (profileData, history) => async (dispatch) => {
+  if (localStorage.token) {
+    setAuthToken(`Bearer ${localStorage.token}`);
+  }
```

Comment:
Create a function outside this file for checking if the token is local storage and call it whenever you need it.